src/events/onReactionAddUpdateStarboard.evt.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2025 Inimi | DrHooBs | Incoverse
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { DrBotEvent, DrBotEventTypeSettings, DrBotEventTypes } from "@src/lib/base/DrBotEvent.js";
+import { Client, Embed, EmbedBuilder, Events, Guild, TextChannel } from "discord.js";
+import { DrBotGlobal } from "@src/interfaces/global.js";
+
+export default class ORAUS extends DrBotEvent {
+    declare global: DrBotGlobal;
+    
+    protected _type: DrBotEventTypes = "discordEvent";
+
+    protected _typeSettings: DrBotEventTypeSettings = {
+        listenerKey: Events.MessageReactionAdd,
+    }
+
+    protected _priority: number = 0;
+    
+
+    public async runEvent(reaction, user, client: Client) {
+        if (global.app.config.starboardEmojiID === "" || global.app.config.starboardEmojiSTR === "" || global.app.config.starboardNumber === 0) {
+            return;
+        }
+        
+        if (reaction.partial) {
+            try {
+                await reaction.fetch();
+                await user.fetch();
+            } catch (error) {
+                console.error("Something went wrong when fetching the message: ", error);
+                return;
+            }
+        }
+
+    
+        if (reaction.message.channel.name.match(/^(staff|mod|admin)([-_][a-z0-9]+)*$/i)) {
```

Comment:
To be honest, I'd say the regex here is really extensive, just checking for "staff" "mod" or "admin" is usually enough, you don't need the rest.

This also won't work since you're starting the regex with ^, meaning start of the line, but as I mentioned in the previous review, the channels don't start on stuff like "staff", it starts with ``»・``, meaning your regex would fail.


```suggestion
        if (reaction.message.channel.name.match(/(staff|mod|admin)/i)) {
```